tree/30_generic_methods/file_from_shared_folder.cf

```diff
@@ -21,9 +21,10 @@
 #
 # @parameter source      Source file (path, relative to /var/rudder/configuration-repository/shared-files)
 # @parameter destination Destination file (absolute path on the target node)
-# @parameter hash_type   Hash algorithm used to check if file is updated (md5, sha1, sha256). Only used on Windows, ignored on Unix.
-# @parameter_constraint hash_type "select" : [ "md5", "sha1", "sha256" ]
-#
+# @parameter hash_type   Hash algorithm used to check if file is updated (sha256, sha512). Only used on Windows, ignored on Unix. default is sha256
```

Comment:
we can't have a default here because it would break windows method and we can't force windows plugin version sync easily

Reply:
(already thought about adding one as it really inconvenient on Linux, should have added a comment here)

Reply:
I think we can update it, we know all users of the plugins and can make them upgrade, it's so much a PITA

Reply:
then you need to actually define the default value in the CFengine code with a defaults promise.

Reply:
we don't need it in cfengine, and it can be empty

Reply:
The value will still be defined but empty

Reply:
indeed, we don't use it with CFEngine